nodestream/project/plugin.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+from typing import List
+
+from ..file_io import LoadsFromYamlFile
+from ..pipeline.scope_config import ScopeConfig
+
+
+@dataclass
+class PluginScope(LoadsFromYamlFile):
+    """A `PluginScope` represents a collection of configuration for a plugin.
+
+    A config is a key value pair object in a nodestream scope including plugins.
+    It contains a collection of configuration key value pairs to be used by the pipelines of a scope.
+    """
+
+    name: str
+    config: ScopeConfig = None
+    targets: List[str] = None
```

Comment:
I wonder if we should store the `targets` as the actual `Target` objects. It does provide some cumbersome api to load the targets after listing the targets. Perhaps this is something for a future PR.

Reply:
Yeah we would have to do some refactoring I think as we would have to pass in the Project to the plugin to get target configuration from the project, but we are also setting the plugin in the project. Not sure it gives us much storing the object here instead of the identifying string.

Reply:
We could load the target schema like we do for the plugins and scope config 🤔

Reply:
I think we can work on this in a future refactor since this is mostly a code cleanup task and doesn't effect the functionality of the current runtime.

Reply:
Works for me